internal/bus/message_pipe.go

```diff
@@ -116,13 +116,9 @@ func (p *MessagePipe) Run(ctx context.Context) {
 				r.Close(ctx)
 			}
 
-			close(p.messageChannel)
-
 			return
 		case m := <-p.messageChannel:
-			p.mu.Lock()
```

Comment:
was this removed because bus has it's own mutex?

Reply:
This was removed as a mutex here isn't required. There is no shared resource that needs to be locked.